File: Exercises/Exercise1/task4.c
#include <pthread.h>
#include <stdio.h>

int i = 0;

void* countUp()
{
    for (int j = 0; j < 1000000; j++) {
        i += 1;
    }
    return NULL;
}

void* countDown()
{
    for (int j = 0; j < 1000000; j++) {
        i -= 1;
    }
    return NULL;
}


int main() {
    pthread_t thread_1;
    pthread_t thread_2;
    pthread_create(&thread_1, NULL, countUp, NULL);
    pthread_create(&thread_2, NULL, countDown, NULL);
    pthread_join(thread_1, NULL);
    pthread_join(thread_2, NULL);
    printf("%d\n", i);
}

File: Exercises/Exercise1/task4.py
from threading import Thread

i = 0

def countUp():
    global i
    for j in range(0, 1000000):
        i += 1

def countDown():
    global i
    for k in range(0, 1000000):
        i -= 1

thread1 = Thread(target = countUp, args = (),)
thread2 = Thread(target = countDown, args = (),)
thread1.start()
thread2.start()

thread1.join()
thread2.join()

print i
